x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -84,7 +84,13 @@ public ExtractedFields detect() {
         checkRequiredFieldsArePresent(fields);
 
         if (fields.isEmpty()) {
-            throw ExceptionsHelper.badRequestException("No compatible fields could be detected in index {}", Arrays.toString(index));
+            Set<String> supportedTypes = new HashSet<>(NUMERICAL_TYPES);
+            if (config.getAnalysis().supportsCategoricalFields()) {
+                supportedTypes.addAll(CATEGORICAL_TYPES);
+            }
+            throw ExceptionsHelper.badRequestException("No compatible fields could be detected in index {}. Supported types are {}.",
```

Comment:
Nice! We should also do the same for the last error message in `removeFieldsWithIncompatibleTypes()`

Reply:
roger